import-export-cli/cmd/k8s/addApi.go

```diff
@@ -0,0 +1,230 @@
+		k8sSaveConfig = false
+	}
+
+
+	byteVal, errMarshal := yaml.Marshal(apiCrd)
+	if errMarshal != nil {
+		utils.HandleErrorAndExit("Error marshal api configmap ", errMarshal)
+	}
+	//write configmap to a temp file
+	tmpFile, err := ioutil.TempFile(os.TempDir(), "apicr-*.yaml")
+	if err != nil {
+		log.Fatal("Cannot create temporary file", err)
+	}
+	defer os.Remove(tmpFile.Name())
+	if _, err = tmpFile.Write(byteVal); err != nil {
+		log.Fatal("Failed to write to temporary file", err)
+	}
+	// Close the file
+	if err := tmpFile.Close(); err != nil {
+		log.Fatal(err)
+	}
+
+	k8sArgs := []string{k8sOperation, "-f", tmpFile.Name(), "-n", flagNamespace}
+	if k8sSaveConfig {
+		k8sArgs = append(k8sArgs, "--save-config")
+	}
+
+	//execute kubernetes command to create or update api from file
+	errAddApi := k8sUtils.ExecuteCommand(k8sUtils.Kubectl, k8sArgs...)
+	if errAddApi != nil {
+		fmt.Println("error configuring API")
+		// delete all configs if any error
+		rollbackConfigs(apiCrd)
+	}
+}
+
+// rollbackConfigs deletes configs defined in the API CR given
+func rollbackConfigs(apiCr *wso2v1alpha2.API) {
+	var rollbackConfMaps []string // configmap names to be deleted
+
+	// swagger configmaps
+	rollbackConfMaps = append(rollbackConfMaps, apiCr.Spec.SwaggerConfigMapName)
+
+	if len(rollbackConfMaps) == 0 {
+		return
+	}
+
+	// execute kubernetes command to delete
+	fmt.Println("Deleting created configs")
+	k8sArgs := []string{k8sUtils.K8sDelete, "cm"}
+	k8sArgs = append(k8sArgs, rollbackConfMaps...)
+
+	delConfErr := k8sUtils.ExecuteCommand(k8sUtils.Kubectl, k8sArgs...)
+	if delConfErr != nil {
+		utils.HandleErrorAndExit("error deleting configmaps of the API: "+apiCr.Name, delConfErr)
+	}
+}
+
+func init() {
+	K8sAddCmd.AddCommand(addApiCmd)
+	//addApiCmd.Flags().StringVarP(&flagApiEndPoint, "apiEndPoint", "a", "", "")
+	addApiCmd.Flags().StringVarP(&flagApiName, "name", "n", "", "Name of the API")
+	addApiCmd.Flags().StringVarP(&flagSwaggerFilePath, "from-file", "f", "",
+		"Path to swagger file")
+	//addApiCmd.Flags().IntVar(&flagReplicas, "replicas", 1, "replica set")
+	addApiCmd.Flags().StringVar(&flagNamespace, "namespace", "", "namespace of API")
+	//addApiCmd.Flags().BoolVarP(&flagOverride, "override", "", false,
+	//	"Property to override the existing docker image with the given name and version")
+	//addApiCmd.Flags().StringVarP(&flagApiVersion, "version", "v", "",
+	//	"Property to override the API version")
+	//addApiCmd.Flags().StringVarP(&flagApiMode, "mode", "m", "",
+	//	fmt.Sprintf("Property to override the deploying mode. Available modes: %v, %v",
+	//		utils.PrivateJetModeConst, utils.SidecarModeConst))
+	//addApiCmd.Flags().StringArrayVarP(&flagEnv, "env", "e", []string{},
+	//	"Environment variables to be passed to deployment")
+	//addApiCmd.Flags().StringVarP(&flagImage, "image", "i", "",
+	//	"Image of the API. If specified, ignores the value of --override")
+	//addApiCmd.Flags().StringVarP(&flagHostname, "hostname", "", "",
```

Comment:
Can we remove the above-commented code?

Reply:
Fixed in 8ac841e8cd6ffe78a452a6255072e39a0a020a17